reana_server/rest/analyses.py

```diff
@@ -380,3 +380,104 @@ def analysis_status(analysis_id):  # noqa
+    r"""Set analysis status.
+    ---
+    put:
+      summary: Set status of an analysis.
+      description: >-
+        This resource reports the status of an analysis.
+        Resource is expecting a analysis UUID.
+      operationId: set_analysis_status
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: organization
+          in: query
+          description: Required. Organization which the worklow belongs to.
+          required: true
+          type: string
+        - name: user
+          in: query
+          description: Required. UUID of workflow owner.
+          required: true
+          type: string
+        - name: analysis_id
+          in: path
+          description: Required. Analysis UUID.
+          required: true
+          type: string
+        - name: status
+          in: body
+          description: Required. New analysis status.
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: >-
+            Request succeeded. Info about an analysis, including the status is
+            returned.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+              organization:
+                type: string
+              status:
+                type: string
+              user:
+                type: string
+          examples:
+            application/json:
+              {
+                "id": "256b25f4-4cfb-4684-b7a8-73872ef455a1",
+                "organization": "default_org",
+                "status": "created",
+                "user": "00000000-0000-0000-0000-000000000000"
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed.
+          examples:
+            application/json:
+              {
+                "message": "Malformed request."
+              }
+        403:
+          description: >-
+            Request failed. User is not allowed to access workflow.
+          examples:
+            application/json:
+              {
+                "message": "User 00000000-0000-0000-0000-000000000000
+                            is not allowed to access workflow
+                            256b25f4-4cfb-4684-b7a8-73872ef455a1"
+              }
+        500:
+          description: >-
+            Request failed. Internal controller error.
+    """
+    try:
+        user = request.args['user']
+        organization = request.args['organization']
+        workflow_id = analysis_id
+        status = request.json
+
+        response, http_response = rwc_api_client.api.set_workflow_status(
+            user=user,
+            organization=organization,
+            workflow_id=workflow_id,
+            status=status).result()
+
+        return jsonify(response), http_response.status_code
+    except KeyError as e:
```

Comment:
Regarding exception handling with Bravado have a look to https://github.com/reanahub/reana-server/issues/18